### ISSUE TITLE
Smudge tool leaves a white trail instead of a transparent one

### ISSUE THREAD

**Issue by lucaspontoexe** [pencil2d/pencil#720]
### Issue Summary
Smudge tool leaves a white trail instead of a transparent one. That can lead to problems if someone uses other background color than white.

### Image Reference
https://puu.sh/wZhYJ/bd24d1e241.png
![image](https://user-images.githubusercontent.com/25611847/28856013-e9f2322a-7716-11e7-928b-b48b7042af32.png)

### Steps to reproduce
1. Change the background to checkerboard on Preferences
2. Draw something on a bitmap layer
3. Drag using smudge tool
4. and there's it

---

**Comment by lucaspontoexe** [comment]
_such an abstract modern art tho_ :P

**Comment by lucaspontoexe** [comment]
![image](https://user-images.githubusercontent.com/25611847/41503826-0f4485da-71b4-11e8-8a18-c5fd086e69bd.png)

I did some tests with the latest nightly build and the result is still the same. This time, I created a layer behind the main one

**Comment by Jose-Moreno** [comment]
@lucaspontoexe Thanks a lot. Then we'll leave this issue open until devs can visit it again and fix it. :smile:

**Comment by davidlamhauge** [comment]
I have spent some time looking into this problem, and it puzzles me that it ever worked, so my first question is: can anyone confirm if smudge ever worked? and when?

**Comment by Jose-Moreno** [comment]
@davidlamhauge It worked quite better in 0.5.4 dev (you know it is that version because it had an additional debug window that was removed) but it still was more akin to Photoshop's "finger painting" where you smudge and apply color (white in this case), but it was still a bit of a WIP, however, somehow, on later versions the behavior worsened and even crashed the software plenty when using it.

![image](https://user-images.githubusercontent.com/4912907/65374765-256ff000-dc53-11e9-8df0-d23ddc08303e.png)

**Comment by davidlamhauge** [comment]
@Jose-Moreno Yeah, but it still smudges with white paint as I see it, and that can't be the wanted behaviour? Is that from the time when it worked?
It should smudge the line in a fashion that blurs it with transparency, imo.

**Comment by Jose-Moreno** [comment]
Yeah the white paint is unwanted.

**Comment by lucaspontoexe** [comment]
I'm not experienced with Qt, but I think the white trail might be related to the source mode that is being used.

I'm pasting a link of all modes, just for convenience https://doc.qt.io/qt-5/qpainter.html#composition-modes

**Comment by CandyFace** [comment]
Fixed in 79ede5042ddf29af993768725b614eb6a0495276
Will be available in next nightly build.

**Comment by Jose-Moreno** [comment]
@CandyFace @scribblemaniac I'm reopening this. I've tested the following build

Development Build Jul 30 2020
commit: b2d8535ea71c9140d78ad6329df64d823e74220e
date: 2020-07-30_12:19:20

Operating System: Windows 10 (10.0)
CPU Architecture: x86_64
Qt Version: 5.12.9

And I get the same result as Lucas as seen here:

![image](https://user-images.githubusercontent.com/4912907/88893816-87eb3280-d20c-11ea-968f-777a562c9c32.png)

**Comment by Kaikogf** [comment]
[REVIEW Jan2021]

I tested with the last build [04Jan2021], and I must say it its a LOT better.

After playing around this issue I came to a conclusion.

The mixing feature provided by the Smudge tool is a lot better. 
In order to avoid the white artefacts while using this tool, we should take into account to have a maximum range value of 45 approximately between the **feather** and the **width** tool for the Smudge tool to work properly without showing the white artefacts.
Meaning that if the **feather tool** displays a value of  99 so the _width tool_ should displays a value higher than 50.
